File: sdk/approute-sdk/javascript/tests/factories/funds.ts
import type {
  FundingMethod,
  FundingMethodsResponse,
  FundingInvoice,
  FundingInvoiceList,
  FundingInvoiceTimeLeft,
  TonDepositState,
  BybitState,
} from '../../src/models/index.js';

/**
 * Helper: JSON responses use `null` for absent optional fields.
 */
type Nullable<T> = { [K in keyof T]: T[K] | null };

export function makeFundingMethod(
  overrides?: Partial<FundingMethod>,
): FundingMethod {
  return {
    code: 'USDT_TRC20',
    name: 'USDT (TRC-20)',
    minAmount: 10.0,
    commission: 0.0,
    address: 'TXyz123abc',
    ttlMinutes: 60,
    confirmationsRequired: 20,
    ...overrides,
  };
}

export function makeFundingMethodsResponse(
  overrides?: Partial<FundingMethodsResponse>,
): FundingMethodsResponse {
  return {
    items: [makeFundingMethod()],
    ...overrides,
  };
}

export function makeFundingInvoice(
  overrides?: Partial<Nullable<FundingInvoice>>,
): FundingInvoice {
  return {
    id: 'inv-001',
    methodCode: 'USDT_TRC20',
    amountExpected: 100.0,
    commission: 0.0,
    credited: 0.0,
    memoTag: null,
    address: 'TXyz123abc',
    txHash: null,
    status: 'pending',
    confirmationsRequired: 20,
    confirmations: 0,
    createdAt: '2026-03-01T12:00:00Z',
    expiresAt: '2026-03-01T13:00:00Z',
    direction: 'incoming',
    ...overrides,
  } as FundingInvoice;
}

export function makeFundingInvoiceList(
  overrides?: Partial<FundingInvoiceList>,
): FundingInvoiceList {
  return {
    items: [makeFundingInvoice()],
    total: 1,
    ...overrides,
  };
}

export function makeFundingInvoiceTimeLeft(
  overrides?: Partial<FundingInvoiceTimeLeft>,
): FundingInvoiceTimeLeft {
  return {
    invoiceId: 'inv-001',
    expiresAt: '2026-03-01T13:00:00Z',
    secondsLeft: 3200,
    expired: false,
    ...overrides,
  };
}

export function makeTonDepositState(
  overrides?: Partial<TonDepositState>,
): TonDepositState {
  return {
    address: 'EQDtFpEwcFAEcRe5mLVh2N6C0x-_hJEM7W61_JLnSF78p7Av',
    memoTag: '123456',
    ...overrides,
  };
}

export function makeBybitState(
  overrides?: Partial<BybitState>,
): BybitState {
  return {
    recipientUid: 'bybit-uid-001',
    linked: true,
    yourUid: 'bybit-uid-002',
    ...overrides,
  };
}

File: sdk/approute-sdk/python/tests/factories/funds.py
from __future__ import annotations

from datetime import datetime, timezone

from approute.models import (
    BybitState,
    FundingInvoice,
    FundingInvoiceList,
    FundingInvoiceTimeLeft,
    FundingMethod,
    FundingMethodsResponse,
    TonDepositState,
)

_CREATED = datetime(2026, 3, 1, 12, 0, 0, tzinfo=timezone.utc)
_EXPIRES = datetime(2026, 3, 1, 13, 0, 0, tzinfo=timezone.utc)


def make_funding_method(
    *,
    code: str = "USDT_TRC20",
    name: str = "USDT (TRC-20)",
    min_amount: float = 10.0,
    commission: float = 0.0,
    address: str = "TXyz123abc",
    ttl_minutes: int = 60,
    confirmations_required: int = 20,
) -> FundingMethod:
    return FundingMethod(
        code=code,
        name=name,
        min_amount=min_amount,
        commission=commission,
        address=address,
        ttl_minutes=ttl_minutes,
        confirmations_required=confirmations_required,
    )


def make_funding_methods_response(
    *,
    methods: list[FundingMethod] | None = None,
) -> FundingMethodsResponse:
    return FundingMethodsResponse(
        items=methods if methods is not None else [make_funding_method()],
    )


def make_funding_invoice(
    *,
    id: str = "inv-001",
    method_code: str = "USDT_TRC20",
    amount_expected: float = 100.0,
    commission: float = 0.0,
    credited: float = 0.0,
    address: str = "TXyz123abc",
    status: str = "pending",
    confirmations_required: int = 20,
    confirmations: int = 0,
    created_at: datetime = _CREATED,
    expires_at: datetime = _EXPIRES,
) -> FundingInvoice:
    return FundingInvoice(
        id=id,
        method_code=method_code,
        amount_expected=amount_expected,
        commission=commission,
        credited=credited,
        address=address,
        status=status,
        confirmations_required=confirmations_required,
        confirmations=confirmations,
        created_at=created_at,
        expires_at=expires_at,
    )


def make_funding_invoice_list(
    *,
    invoices: list[FundingInvoice] | None = None,
    total: int = 1,
) -> FundingInvoiceList:
    return FundingInvoiceList(
        items=invoices if invoices is not None else [make_funding_invoice()],
        total=total,
    )


def make_funding_invoice_time_left(
    *,
    invoice_id: str = "inv-001",
    expires_at: datetime = _EXPIRES,
    seconds_left: int = 3200,
    expired: bool = False,
) -> FundingInvoiceTimeLeft:
    return FundingInvoiceTimeLeft(
        invoice_id=invoice_id,
        expires_at=expires_at,
        seconds_left=seconds_left,
        expired=expired,
    )


def make_ton_deposit_state(
    *,
    address: str = "EQDtFpEwcFAEcRe5mLVh2N6C0x-_hJEM7W61_JLnSF78p7Av",
    memo_tag: str = "123456",
) -> TonDepositState:
    return TonDepositState(address=address, memo_tag=memo_tag)


def make_bybit_state(
    *,
    recipient_uid: str = "bybit-uid-001",
    linked: bool = True,
    your_uid: str | None = "bybit-uid-002",
) -> BybitState:
    return BybitState(recipient_uid=recipient_uid, linked=linked, your_uid=your_uid)
